From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-show_mem-add-trylock-while-printing-alloc-info-fix
Date: Tue Sep  2 08:22:40 PM PDT 2025

fix build with CONFIG_MEM_ALLOC_PROFILING=n

mm/show_mem.c: In function '__show_mem':
mm/show_mem.c:403:32: error: unused variable 'mem_alloc_profiling_spinlock' [-Werror=unused-variable]
  403 |         static DEFINE_SPINLOCK(mem_alloc_profiling_spinlock);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~

Cc: Brendan Jackman <jackmanb@google.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Usama Arif <usamaarif642@gmail.com>
Cc: Vishal Moola (Oracle) <vishal.moola@gmail.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Cc: Yueyang Pan <pyyjason@gmail.com>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/show_mem.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/show_mem.c~mm-show_mem-add-trylock-while-printing-alloc-info-fix
+++ a/mm/show_mem.c
@@ -400,7 +400,9 @@ static void show_free_areas(unsigned int
 
 void __show_mem(unsigned int filter, nodemask_t *nodemask, int max_zone_idx)
 {
+#ifdef CONFIG_MEM_ALLOC_PROFILING
 	static DEFINE_SPINLOCK(mem_alloc_profiling_spinlock);
+#endif
 	unsigned long total = 0, reserved = 0, highmem = 0;
 	struct zone *zone;
 
_
